saga/src/main/java/io/axoniq/dev/samples/saga/ProcessOrderSaga.java

```diff
@@ -67,6 +70,7 @@ public void on(OrderConfirmedEvent event, DeadlineManager deadlineManager, UUIDP
 
         //This order should be completed in 5 days
         this.orderDeadlineId = deadlineManager.schedule(Duration.of(5, ChronoUnit.DAYS), ORDER_COMPLETE_DEADLINE);
+        axonServerEventScheduler.schedule(Duration.of(5, ChronoUnit.DAYS), new SagaEndedEvent(orderId));
```

Comment:
I understand why you've added the scheduler here as you did, but I do think it blocks/mixes the concerns this deadline-specific method currently fulfills.
As far as samples go, I'd either (1) see the benefit in an entirely different sample from this Saga package or (2) a separate event handling flow just for this use case within this sample.